### PR TITLE
Add missing hip workgroup unordered policy

### PR DESCRIPTION
**Pull request by MrBurmark**
# Summary
Forgot to add this policy for unordered hip workgroups or it was removed by accident.

- This PR is a bugfix
- It does the following (modify list as needed):
  - Fixes missing policy

### CODE REVIEW

**Review by artv3 — APPROVED** [review]

File: include/RAJA/policy/hip/policy.hpp [hunk 1]
```diff
@@ -94,6 +94,15 @@ struct hip_work : public RAJA::make_policy_pattern_launch_platform_t<
                        RAJA::Platform::hip> {
 };
 
+#if defined(RAJA_ENABLE_HIP_INDIRECT_FUNCTION_CALL)
+struct unordered_hip_loop_y_block_iter_x_threadblock_average
```
Comment (rhornung67) [inline include/RAJA/policy/hip/policy.hpp:98]: Wow! That is some policy name.

**Review by rhornung67 — APPROVED** [review]

File: include/RAJA/policy/hip/policy.hpp (hunk 1 above)
Comment (MrBurmark) [reply include/RAJA/policy/hip/policy.hpp:98]: I tried to make it as descriptive as possible. Its a copy of the cuda version.

Comment (MrBurmark) [reply include/RAJA/policy/hip/policy.hpp:98]: Its trying to say that loops or for_alls are mapped onto blocks in the y dimension and iterations in the loops are mapped to threadblocks in the x dimension.

Comment (rhornung67) [reply include/RAJA/policy/hip/policy.hpp:98]: got it.

Comment (artv3) [reply include/RAJA/policy/hip/policy.hpp:98]: > Its trying to say that loops or for_alls are mapped onto blocks in the y dimension and iterations in the loops are mapped to threadblocks in the x dimension.

That's pretty much what I had in mind when I first read it :grinning:

**Pull request merged**